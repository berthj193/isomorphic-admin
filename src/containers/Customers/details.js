import React from 'react';
import PropTypes from 'prop-types';
import Address from './address';
import customersQueries from '../../graphql/customers';
import { getListFor } from '../../graphql/orders';
import PageLoading from '../PageLoading/PageLoading';
import DetailPage from '../../components/details/details';
import ItemList from '../../components/list/item-list';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';
import withOrdersManager from '../../components/apolloProviders/withOrdersManager';
import orderColumns from '../Orders/columns';
import { toListItem } from '../../mappers/order';

export class CustomerDetails extends React.Component {
  static propTypes = {
    auth0_id: PropTypes.string,
    customers: PropTypes.arrayOf(PropTypes.shape({
      given_name: PropTypes.string,
      family_name: PropTypes.string,
      email: PropTypes.string,

      payment_sources: PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
          brand: PropTypes.string.isRequired,
          created_at: PropTypes.string.isRequired,
          last4: PropTypes.string,
          payment_method: PropTypes.string,
        }),
      ),
    })),
    orders: PropTypes.arrayOf(PropTypes.object),
    error: PropTypes.string,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string,
      }),
    }).isRequired,
  };

  static defaultProps = {
    customers: [],
    orders: [],
  };

  // Temporary solution. No way to get single customer yet
  getCustomer() {
    return this.props.customers
      .find(({ auth0_id }) => auth0_id === this.props.match.params.id);
  }

  loadDetailList = () => {
    const {
      given_name,
      family_name,
      email,
      addresses,
      payment_sources,
    } = this.getCustomer();

    return [
      { Name: `${given_name} ${family_name}` },
      { Email: email },
      '---',
      { Addresses: () => addresses.map((address, index) => <Address {...address} key={index} />) },
      { 'Payment Methods': payment_sources },
    ];
  };

  render() {
    if (!this.props.error && !this.getCustomer()) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }

    return (
      <React.Fragment>
        <LayoutContentWrapper style={{ minHeight: '100vh' }}>
          <LayoutContent>
            <h1>Customer</h1>
            <ButtonList
              buttons={[
                { link: '/dashboard/customers', title: 'Go back' },
              ]} />
            <DetailPage details={this.loadDetailList()} />
            <ItemList
              data={toListItem(this.props.orders)}
              title="Customer Orders"
              columns={orderColumns}
              detailsPath="/dashboard/orders"
            />
          </LayoutContent>
        </LayoutContentWrapper>
      </React.Fragment>
    );
  }
}

export default withOrdersManager(CustomerDetails, ({ match: { params: { id } } }) => [
  {
    query: customersQueries.getList(),
  },
  {
    query: getListFor,
    variables: { auth0Id: id },
  },
]);
