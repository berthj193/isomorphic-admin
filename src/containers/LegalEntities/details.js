import React from 'react';
import PropTypes from 'prop-types';
import { getList } from '../../graphql/legalEntities';
import PageLoading from '../PageLoading/PageLoading';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import DetailPage from '../../components/details/details';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

export class LegalEntityDetails extends React.Component {
  static propTypes = {
    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    legal_entities: PropTypes.arrayOf(PropTypes.object),
    error: PropTypes.string,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string,
      }),
    }),
  };

  static defaultProps = {
    legal_entities: [{}],
  };

  legalEntityId = () => this.props.match.params.id;

  legalEntity = () =>
    this.props.legal_entities.find(({ id }) => id === this.legalEntityId()) || {};

  loadDetailList = () => {
    const {
      name,
      vat_number,
      stores = [],
      updated_at,
    } = this.legalEntity();

    return [
      { Name: name },
      { 'VAT number': vat_number },
      '---',
      { Stores: this.renderList(stores.map(store => store.key)) },
      '---',
      { 'Updated at': updated_at },
    ];
  };

  renderList = data => () =>
    <ul className="detail-list">
      { data.map(item =>
        <li className="detail-list-item" key={item}>{item}</li>) }
    </ul>;

  render() {
    if (!this.props.error && !this.legalEntity().id) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }

    const { match: { params: { id } } } = this.props;

    return (
      <React.Fragment>
        <LayoutContentWrapper style={{ minHeight: '100vh' }}>
          <LayoutContent>
            <h1>Legal entities</h1>
            <ButtonList
              buttons={[
                { link: '/dashboard/legal_entities', title: 'Go back' },
                { link: `/dashboard/legal_entities/${id}/edit`, title: 'Edit' },
              ]} />
            <DetailPage details={this.loadDetailList()} />
          </LayoutContent>
        </LayoutContentWrapper>
      </React.Fragment>
    );
  }
}

export default withStoresManager(LegalEntityDetails,
  ({ match: { params: { id } } }) => ({
    query: getList,
    variables: {
      id,
    },
  }),
);
