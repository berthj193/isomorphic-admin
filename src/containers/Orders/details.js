import React from 'react';
import PropTypes from 'prop-types';
import { getDetails } from '../../graphql/orders';
import PageLoading from '../PageLoading/PageLoading';
import withOrdersManager from '../../components/apolloProviders/withOrdersManager';
import DetailPage from '../../components/details/details';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

export class OrderDetails extends React.Component {
  static propTypes = {
    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    orders: PropTypes.arrayOf(PropTypes.object),
    error: PropTypes.string,
    match: PropTypes.object,
  };

  static defaultProps = {
    orders: [{}],
  };

  loadDetailList = () => {
    const {
      customer: { given_name, family_name, email, phone } = {},
      store_key,
      shipping_address: {
        city, region, line_one, line_two,
        postcode, given_name: shippingGivenName, family_name: shippingFamilyName,
      } = {},
      item_total = {},
      grand_total = {},
      shipping_total = {},
      item_discount_total = {},
      shipping_discount_total = {},
    } = this.props.orders[0] || {};

    return [
      { Store: store_key },
      '---',
      { 'E-mail': email },
      { 'First name': given_name },
      { 'Last name': family_name },
      { 'Phone number': phone },
      '---',
      { 'First name': shippingGivenName },
      { 'Last name': shippingFamilyName },
      { City: city },
      { Region: region },
      { Postcode: postcode },
      { 'Line one': line_one },
      { 'Line two': line_two },
      '---',
      { 'Item total': `${item_total.value} ${item_total.currency}` },
      { 'Grand total': `${grand_total.value} ${grand_total.currency}` },
      { 'Shipping total': `${shipping_total.value} ${shipping_total.currency}` },
      { 'Item discount total': `${item_discount_total.value} ${item_discount_total.currency}` },
      { 'Shipping discount total': `${shipping_discount_total.value} ${shipping_discount_total.currency}` },
    ];
  };

  render() {
    if (!this.props.error && !this.props.orders[0].id) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }

    const { match: { params: { key } } } = this.props;

    return (
      <React.Fragment>
        <LayoutContentWrapper style={{ minHeight: '100vh' }}>
          <LayoutContent>
            <h1>Orders</h1>
            <ButtonList
              buttons={[
                { link: '/dashboard/orders', title: 'Go back' },
                { link: `/dashboard/orders/${key}/refund`, title: 'Add refund' },
              ]} />
            <DetailPage details={this.loadDetailList()} />
          </LayoutContent>
        </LayoutContentWrapper>
      </React.Fragment>
    );
  }
}

export default withOrdersManager(OrderDetails,
  ({ match: { params: { key } } }) => ({
    query: getDetails,
    variables: {
      id: key,
    },
  }),
);
