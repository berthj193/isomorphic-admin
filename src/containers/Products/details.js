import React from 'react';
import PropTypes from 'prop-types';
import _get from 'lodash/get';

import productQueries from '../../graphql/products';
import PageLoading from '../PageLoading/PageLoading';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import DetailPage from '../../components/details/details';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';
import Error from '../../components/error/error';

export class ProductDetails extends React.Component {
  static propTypes = {
    products: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string,
      default_master_sku: PropTypes.string,
      default_name: PropTypes.string,
      defaultDescription: PropTypes.string,
    })),
    error: PropTypes.string,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string,
      }),
    }),
  };

  static defaultProps = {
    products: [],
  };

  getProductId() {
    return _get(this.props, 'match.params.id');
  }

  // Temporary solution. No way to get single product yet
  getProduct() {
    const productId = this.getProductId();
    return this.props.products.find(({ id }) => id === productId);
  }

  loadDetailList = () => {
    const {
      default_master_sku,
      default_name,
      default_description,
    } = this.getProduct();

    return [
      { Name: default_name },
      { Description: default_description },
      '---',
      { SKU: default_master_sku },
    ];
  };

  render() {
    if (!this.props.error && !this.getProduct()) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Product</h1>
          <ButtonList
            buttons={[
              { link: '/dashboard/products', title: 'Go back' },
              { link: `/dashboard/products/${this.getProductId()}/edit`, title: 'Edit' },
            ]} />
          <DetailPage details={this.loadDetailList()} />
          <Error error={this.props.error} />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(ProductDetails, {
  query: productQueries.getProducts(),
} );
