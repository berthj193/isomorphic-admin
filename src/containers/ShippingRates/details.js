import React from 'react';
import PropTypes from 'prop-types';
import { getDetails } from '../../graphql/shippingRates';
import PageLoading from '../PageLoading/PageLoading';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import DetailPage from '../../components/details/details';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

export class ShippingRateDetails extends React.Component {
  static propTypes = {
    shipping_rates: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        country: PropTypes.string,
        name: PropTypes.string,
        premium: PropTypes.bool,
        live: PropTypes.bool,
        updated_at: PropTypes.string,
      }),
    ),
    error: PropTypes.string,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string,
      }),
    }),
  };

  static defaultProps = {
    shipping_rates: [{}],
  };

  getShippingRate() {
    return this.props.shipping_rates.find(({ id }) => id === this.props.match.params.id);
  }

  loadDetailList = () => {
    const {
      country,
      name,
      premium,
      live,
      updated_at,
    } = this.getShippingRate() || {};

    return [
      { Country: country },
      { Name: name },
      '---',
      { 'Is premium': premium ? 'Yes' : 'No' },
      { 'Is live': live ? 'Yes' : 'No' },
      '---',
      { 'Updated at': updated_at },
    ];
  };

  render() {
    if (!this.props.error && !this.getShippingRate()) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }

    const { match: { params: { id } } } = this.props;

    return (
      <React.Fragment>
        <LayoutContentWrapper style={{ minHeight: '100vh' }}>
          <LayoutContent>
            <h1>Warehouses</h1>
            <ButtonList
              buttons={[
                { link: '/dashboard/shipping_rates', title: 'Go back' },
                { link: `/dashboard/shipping_rates/${id}/edit`, title: 'Edit' },
              ]} />
            <DetailPage details={this.loadDetailList()} />
          </LayoutContent>
        </LayoutContentWrapper>
      </React.Fragment>
    );
  }
}

export default withStoresManager(ShippingRateDetails,
  ({ match: { params: { id } } }) => ({
    query: getDetails,
    variables: {
      id,
    },
  }),
);
