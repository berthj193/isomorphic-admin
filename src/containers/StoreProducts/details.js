import React from 'react';
import PropTypes from 'prop-types';
import storeProductQueries from '../../graphql/storeProducts';
import PageLoading from '../PageLoading/PageLoading';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import DetailPage from '../../components/details/details';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

export class StoreProductDetails extends React.Component {
  static propTypes = {
    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    store_products: PropTypes.arrayOf(PropTypes.object),
    error: PropTypes.string,
    match: PropTypes.object,
  };

  static defaultProps = {
    store_products: [{}],
  };

  loadDetailList = () => {
    const {
      name, master_sku, description,
      product: { default_name: product } = {},
      store: { key: store } = {},
      product_variants = [],
    } = this.props.store_products[0] || {};

    return [
      { Name: name },
      { 'Master SKU': master_sku },
      '---',
      { Product: product },
      { Store: store },
      '---',
      { Description: description },
      { 'Product variants': product_variants.map(({ name }) => name).join(', ') },
    ];
  };

  render() {
    if (!this.props.error && !this.props.store_products[0].id) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }

    const { match: { params: { store_key, id } } } = this.props;

    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Store Products</h1>
          <ButtonList
            buttons={[
              { link: '/dashboard/store_products', title: 'Go back' },
              { link: `/dashboard/store_products/${store_key}/${id}/edit`, title: 'Edit' },
            ]} />
          <DetailPage details={this.loadDetailList()} />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(StoreProductDetails,
  ({ match: { params: { store_key } } }) => ({
    query: storeProductQueries.getDetails,
    variables: {
      store_key,
    },
  }),
);
