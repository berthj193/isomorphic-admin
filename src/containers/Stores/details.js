import React from 'react';
import PropTypes from 'prop-types';

import _get from 'lodash/get';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';
import Error from '../../components/error/error';
import DetailPage from '../../components/details/details';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import storeQueries from '../../graphql/stores';

export class StoreDetails extends React.Component {
  static propTypes = {
    id: PropTypes.number,
    store: PropTypes.object,
    error: PropTypes.string,
    stores: PropTypes.arrayOf(PropTypes.object),
  };

  static defaultProps = {
    stores: [],
  };

  getStoreKey() {
    return _get(this.props, 'match.params.key');
  }

  loadDetailList = () => {
    const {
      key,
      warehouse: { name: warehouseName } = {},
      channel: { name: channelName } = {},
      legal_entity: { name: legalEntityName } = {},
      storeProducts,
      created_at,
    } = this.props.stores[0] || {};

    return [
      { Key: key },
      '---',
      { Warehouse: warehouseName },
      { Channel: channelName },
      { 'Legal entity': legalEntityName },
      { 'Store products': storeProducts },
      { 'Created at': created_at },
    ];
  };

  render() {
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Stores</h1>
          <ButtonList
            buttons={[
              { link: '/dashboard/stores', title: 'Go back' },
              { link: `/dashboard/stores/${this.getStoreKey()}/edit`, title: 'Edit' },
            ]} />
          <DetailPage details={this.loadDetailList()} />
          <Error error={this.props.error} />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(
  StoreDetails,
  ({ match: { params: { key } } }) => (
    {
      query: storeQueries.getDetails,
      variables: {
        store_key: key,
      },
    }
  ),
);
