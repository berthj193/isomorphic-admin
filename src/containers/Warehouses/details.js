import React from 'react';
import PropTypes from 'prop-types';
import { getDetails } from '../../graphql/warehouses';
import PageLoading from '../PageLoading/PageLoading';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import DetailPage from '../../components/details/details';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

export class WarehouseDetails extends React.Component {
  static propTypes = {
    warehouses: PropTypes.arrayOf(PropTypes.object),
    error: PropTypes.string,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string,
      }),
    }),
  };

  static defaultProps = {
    warehouses: [],
  };

  getWarehouse = () =>
    this.props.warehouses[0] || null;

  loadDetailList = () => {
    const {
      name,
      export_class,
      options = [],
      shipping_rates = [],
      stores = [],
      updated_at,
    } = this.getWarehouse() || {};
    return [
      { Name: name },
      { 'Export class': export_class },
      '---',
      options.length > 0 && { Options: this.renderOptions(options) },
      options.length > 0 && '---',
      shipping_rates.length > 0 && { 'Shipping rates': this.renderShippingRates(shipping_rates) },
      shipping_rates.length > 0 && '---',
      stores.length > 0 && { Stores: this.renderStores(stores) },
      stores.length > 0 && '---',
      { 'Updated at': updated_at },
    ];
  };

  renderOptions = data => () =>
    data.map(option => (
      <ul className="key-value-detail" key={option.key}>
        <li>
          <span className="key-value-detail-title">{option.key}</span>
          <span className="key-value-detail-value">{option.value}</span>
        </li>
      </ul>
    ));

  renderStores = stores => this.renderList(stores.map(({ key }) => key));

  renderShippingRates = rates => this.renderList(rates.map(({ name }) => name));

  renderList = data => () =>
    <ul className="detail-list">
      { data.map(value => <li className="detail-list-item" key={value}>{value}</li> ) }
    </ul>;

  render() {
    if (!this.props.error && !this.getWarehouse()) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }

    const { match: { params: { id } } } = this.props;

    return (
      <React.Fragment>
        <LayoutContentWrapper style={{ minHeight: '100vh' }}>
          <LayoutContent>
            <h1>Warehouses</h1>
            <ButtonList
              buttons={[
                { link: '/dashboard/warehouses', title: 'Go back' },
                { link: `/dashboard/warehouses/${id}/edit`, title: 'Edit' },
              ]} />
            <DetailPage details={this.loadDetailList()} />
          </LayoutContent>
        </LayoutContentWrapper>
      </React.Fragment>
    );
  }
}

export default withStoresManager(WarehouseDetails,
  ({ match: { params: { id } } }) => ({
    query: getDetails,
    variables: {
      id,
    },
  }),
);
