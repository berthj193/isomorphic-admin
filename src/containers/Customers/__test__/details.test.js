import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { CustomerDetails } from '../details';

describe('<CustomerDetails />', () => {
  const defaultProps = {
    match: {
      params: {
        id: '1',
      },
    },
  };

  it('should render loading if there is no customer to display', () => {
    expect(toJson(shallow(<CustomerDetails {...defaultProps} />))).toMatchSnapshot();
  });

  it('should render properly, if customer has been found', () => {
    const props = {
      customers: [
        {
          auth0_id: 'holonet|d3str0y-d3ath-st4r',
          given_name: 'Leia',
          family_name: 'Organa',
          email: 'lorgana@rebellion.ga',
        },
        {
          auth0_id: 'imperialnet|1-am-th3-s3n4t3',
          given_name: 'Sheev',
          family_name: 'Palpatine',
          email: 'palpi@coruscant.co',
        },
      ],
      match: {
        params: {
          id: 'holonet|d3str0y-d3ath-st4r',
        },
      },
    };

    expect(toJson(shallow(<CustomerDetails {...{ ...defaultProps, ...props }} />))).toMatchSnapshot();
  });
});
