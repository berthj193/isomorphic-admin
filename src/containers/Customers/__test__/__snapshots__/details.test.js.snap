// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerDetails /> should render loading if there is no customer to display 1`] = `
<PageLoading
  style={
    Object {
      "height": "50vh",
    }
  }
/>
`;

exports[`<CustomerDetails /> should render properly, if customer has been found 1`] = `
<React.Fragment>
  <LayoutWrapper
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <LayoutContent>
      <h1>
        Customer
      </h1>
      <ButtonList
        buttons={
          Array [
            Object {
              "link": "/dashboard/customers",
              "title": "Go back",
            },
          ]
        }
      />
      <PageDetails
        details={
          Array [
            Object {
              "Name": "Leia Organa",
            },
            Object {
              "Email": "lorgana@rebellion.ga",
            },
            "---",
            Object {
              "Addresses": [Function],
            },
            Object {
              "Payment Methods": undefined,
            },
          ]
        }
      />
      <withRouter(ItemList)
        columns={
          Array [
            Object {
              "dataIndex": "id",
              "key": "id",
              "title": "ID",
            },
            Object {
              "dataIndex": "customer_name",
              "key": "customer_name",
              "title": "Name",
            },
            Object {
              "dataIndex": "email",
              "key": "email",
              "title": "Email",
            },
            Object {
              "dataIndex": "created_at",
              "key": "created_at",
              "title": "Ordered At",
            },
            Object {
              "dataIndex": "status",
              "key": "status",
              "title": "Status",
            },
            Object {
              "dataIndex": "grand_total",
              "key": "grand_total",
              "title": "Order Total",
            },
          ]
        }
        data={Array []}
        detailsPath="/dashboard/orders"
        title="Customer Orders"
      />
    </LayoutContent>
  </LayoutWrapper>
</React.Fragment>
`;
